connectors/sources/sharepoint_online.py

```diff
@@ -1750,38 +1758,49 @@ async def get_docs_incrementally(self, sync_cursor, filtering=None):
                         yield list_item, download_func, OP_INDEX
 
                 # Sync site pages
-                async for site_page in self.site_pages(site, site_access_control):
+                async for site_page in self.site_pages(
+                    site, site_access_control, check_timestamp=True
+                ):
                     # Always include site admins in site page access controls
                     site_page = self._decorate_with_access_control(
                         site_page, site_admin_access_control
                     )
                     yield site_page, None, OP_INDEX
 
+        self.update_sync_timestamp_cursor(timestamp)
```

Comment:
Should we also update the `timestamp_cursor` when a full sync job is done?

Reply:
I'm actually updating it when a full sync starts. This is also updating the cursor to the value at the _start_ of the sync, but I have to wait till the end to update it, so that I don't throw things off.

Reply:
Yeah, you are updating it when `sync_cursor` is empty. I'm thinking of a use case when:

1.  you run a full sync -> sync_cursor is initialized with the timestamp
2. you run a couple of incremental syncs -> sync cursor is updated as well
3. you run a full sync again -> sync cursor is not updated
4. you run an incremental sync and it uses the timestamp from the step 2 not the step 3